semver.js

```diff
@@ -704,6 +704,42 @@ Comparator.prototype.test = function(version) {
   return cmp(version, this.operator, this.semver, this.loose);
 };
 
+Comparator.prototype.intersects = function(compOrStr, loose, platform) {
+  var rangeTmp;
+  var comp = new Comparator(compOrStr, loose);
```

Comment:
I am ok with this change, but I actually do not think the user makes more decisions. Since we allow the method to receive either a comparator or a string they can do all the same decisions if they want.

But the API is definitely more explicit if we force a Comparator.

Usually I am a bit more defensive and that is why I usually try to handle all the weird possibilities. But if this was Typescript it would make much more sense to receive only the comparator.